packages/astro/src/core/build/static-build.ts

```diff
@@ -125,6 +148,22 @@ async function ssrBuild(opts: StaticBuildOptions, internals: BuildInternals, inp
 					assetFileNames: 'assets/[name].[hash][extname]',
 					...viteConfig.build?.rollupOptions?.output,
 					entryFileNames: opts.buildConfig.serverEntry,
+					manualChunks: opts.settings.config.experimental.prerender ? (id, api) => {
+						// Split the Astro runtime into a separate chunk for readability
+						if (id.includes('astro/dist')) {
+							return 'astro';
+						}
+						const pageInfo = internals.pagesByViteID.get(id);
+						if (pageInfo) {
+							// prerendered pages should be split into their own chunk
+							// Important: this can't be in the `pages/` directory!
+							if (api.getModuleInfo(id)?.meta.astro?.pageOptions?.prerender) {
+								return `prerender`;
+							}
+							// pages should go in their own chunks/pages/* directory
+							return `pages${pageInfo.route.route.replace(/\/$/, '/index')}`;
+						}
+					} : undefined,
```

Comment:
Fairly substantial change that's more future-facing: we're being more prescriptive about the build output. This is much closer to a one-route-per-file output that might be useful in the future.